src/dmd/lexer.d

```diff
@@ -145,6 +145,24 @@ class Lexer
         }
     }
 
+    version (DMDLIB)
+    {
+        bool empty()
+        {
+            return front() == TOK.endOfFile;
+        }
+
+        TOK front() const @property
```

Comment:
Can you anonate this and the rest of the functions?

Reply:
Do you mean to add `@safe nothrow @nogc pure` ?